consensus/consensus_service.go

```diff
@@ -308,7 +251,7 @@ func (consensus *Consensus) checkViewID(msg *FBFTMessage) error {
 		consensus.current.SetViewID(msg.ViewID)
 		consensus.LeaderPubKey = msg.SenderPubkey
```

Comment:
Better add a func for updating the leaderPubKey with the logic to update IsLeader. There are many places LeaderPubKey is updated without updated the IsLeader.

Reply:
yes, I had that on bigger PR but then there are issues about Getconsensus private key and whatnot, will address in later PR